src/gt4py/next/otf/compilation/build_systems/compiledb.py

```diff
@@ -321,14 +321,16 @@ def _cc_create_compiledb(
     assert compile_db
 
     for entry in compile_db:
-        entry["directory"] = entry["directory"].replace(str(path), "$SRC_PATH")
+        entry["directory"] = entry["directory"].replace(str(path), ".")
         entry["command"] = (
             entry["command"]
             .replace(f"CMakeFiles/{name}.dir", ".")
             .replace(str(path), "$SRC_PATH")
             .replace(binding_src_name, "$BINDINGS_FILE")
             .replace(name, "$NAME")
             .replace("-I$SRC_PATH/build/_deps", f"-I{path}/build/_deps")
+            # make path relative to the build folder
+            .replace("$SRC_PATH/$BINDINGS_FILE", "../$BINDINGS_FILE")
```

Comment:
This whole expression is hardly understandable. For example, `"SRC_PATH"` is implicitly replaced by different values (`{path}`, `..`), but `str(path)` was replaced first by `$SRC_PATH` first just to replace back `$SRC_PATH` to `path` in the include path directive. Also, in the "directory" entry we replace the old `"SRC_PATH"` by `.` but here it is replaced by `..` in the replacement. Are we sure we can't simplify it?

Reply:
I simplified the logic and made it more robust (I believe).